up42/catalog.py

```diff
@@ -263,23 +262,21 @@ def download_quicklooks(
 
         out_paths: List[str] = []
         for image_id in tqdm(image_ids):
-            out_path = output_directory / f"quicklook_{image_id}.jpg"
-            out_paths.append(str(out_path))
-
-            url = (
-                f"{self.auth._endpoint()}/catalog/{provider}/image/{image_id}/quicklook"
-            )
             try:
+                url = f"{self.auth._endpoint()}/catalog/{provider}/image/{image_id}/quicklook"
+
                 response = self.auth._request(
                     request_type="GET", url=url, return_text=False
                 )
-                response.raise_for_status()
-            except HTTPError as err:
-                raise SystemExit(err)
-
-            with open(out_path, "wb") as dst:
-                for chunk in response:
-                    dst.write(chunk)
+                out_path = output_directory / f"quicklook_{image_id}.jpg"
+                out_paths.append(str(out_path))
+                with open(out_path, "wb") as dst:
+                    for chunk in response:
+                        dst.write(chunk)
+            except ValueError:
```

Comment:
Which line from the above try: block would trigger the ValueError? Probably only that line should be in the try block to avoid other unforseen Valueerrors.

Reply:
it is line 268, where else do you think the try mechanism we would get valueerror? SO we thought first we will try to get the quick looks, if they exist, we then write them to the file otherwise we need to skip that iteration of for loop.

Reply:
I see, didn't think about that the raise is gone and this means the whole block needs to be in try/except, ty.

Reply:
Guess best solution to communicate where the valueerror applies & avoid other Valueerrors triggering the message would be like this:

        for image_id in tqdm(image_ids):
            url = f"{self.auth._endpoint()}/catalog/{provider}/image/{image_id}/quicklook"
            try:
                response = self.auth._request(
                    request_type="GET", url=url, return_text=False
                )
            except ValueError:
                logger.warning(
                    f"Image with id {image_id} does not have quicklook available. Skipping ..."
                )
                continue

            out_path = output_directory / f"quicklook_{image_id}.jpg"
            out_paths.append(str(out_path))
            with open(out_path, "wb") as dst:
                for chunk in response:
                dst.write(chunk)

Reply:
shit, I merged the branch, if you want I can do it like this, sorry I though you are also fine with the implemetation

Reply:
Your call, if you think an improvement would be good to add.

Reply:
yeah, maybe when we want to revisit error handling and throwing exceptions, we can look into this again.